vm/src/vm/vm_core.rs

```diff
@@ -910,6 +910,11 @@ impl VirtualMachine {
         self.trace = None
     }
 
+    // TODO: add test cases
```

Comment:
Yes, please :)

Reply:
I actually added this before when this function was doing something else and forgot to remove it when I changed the function. Right now, the function just calls `relocate_segments` which already has test cases.

However, I did add test cases now for `get_relocated_instruction_locations` which I have added in `Program`.